### PR TITLE
Modernize first pass

### PR DESCRIPTION
**Pull request by hjmjohnson**
Apply common C++11 preferred coding approaches.

### CODE REVIEW

File: Modules/Loadable/Texts/qSlicerTextsReader.h [hunk 1]
```diff
@@ -32,13 +32,13 @@ class qSlicerTextsReader : public qSlicerFileReader
 public:
   typedef qSlicerFileReader Superclass;
   qSlicerTextsReader(QObject* parent = 0);
-  virtual ~qSlicerTextsReader();
+  ~qSlicerTextsReader() override override;
```
Comment (jcfr) [inline Modules/Loadable/Texts/qSlicerTextsReader.h]: Look like the keyword is duplicated

File: Modules/Loadable/VolumeRendering/MRML/vtkMRMLShaderPropertyNode.h [hunk 2]
```diff
@@ -58,28 +58,28 @@ class VTK_SLICER_VOLUMERENDERING_MODULE_MRML_EXPORT vtkMRMLShaderPropertyNode
   //--------------------------------------------------------------------------
   // MRMLNode methods
   //--------------------------------------------------------------------------
-  virtual vtkMRMLNode* CreateNodeInstance() VTK_OVERRIDE;
+  vtkMRMLNode* CreateNodeInstance() VTK_OVERRIDE;
```
Comment (jcfr) [inline Modules/Loadable/VolumeRendering/MRML/vtkMRMLShaderPropertyNode.h]: `VTK_OVERRIDE` should probably be removed and replaced with `override`

File: Modules/Loadable/VolumeRendering/qSlicerShaderPropertyReader.h [hunk 3]
```diff
@@ -38,17 +38,17 @@ class qSlicerShaderPropertyReader
   typedef qSlicerFileReader Superclass;
   qSlicerShaderPropertyReader(QObject* parent = nullptr);
   qSlicerShaderPropertyReader(vtkSlicerVolumeRenderingLogic* logic, QObject* parent = nullptr);
-  virtual ~qSlicerShaderPropertyReader();
+  ~qSlicerShaderPropertyReader() override override;
```
Comment (jcfr) [inline Modules/Loadable/VolumeRendering/qSlicerShaderPropertyReader.h]: duplicate

File: Base/QTCLI/vtkSlicerCLIModuleLogic.cxx [hunk 4]
```diff
@@ -1241,18 +1241,18 @@ void vtkSlicerCLIModuleLogic::ApplyTask(void *clientdata)
   // Add a command line flag for the process information structure
   if ( commandType == SharedObjectModule )
     {
-    commandLineAsString.push_back( "--processinformationaddress" );
+    commandLineAsString.emplace_back("--processinformationaddress" );
```
Comment (jcfr) [inline Base/QTCLI/vtkSlicerCLIModuleLogic.cxx]: while updating the file, remove extra space at the end

```
commandLineAsString.emplace_back("--processinformationaddress" );
```

->

```
commandLineAsString.emplace_back("--processinformationaddress");
```

**Comment by jcfr** [comment]
Thanks @hjmjohnson , it looks like commit https://github.com/Slicer/Slicer/pull/4728/commits/9c0164d2474779478938cb1256cb3b0675f789c8 introduce `override override` which is letter fixed in the unrelated commit https://github.com/Slicer/Slicer/pull/4728/commits/6f246738d580c684e751ced4ce30bd5d852e1263.

Could you updates the commit and force push the topic ? :pray:

**Comment by hjmjohnson** [comment]
@jcfr All requested changes have been made.  Local builds completed successfully.

**Comment by jcfr** [comment]
Thanks for updating the commits 🙏 

> re: emplace_back

Since we are not using `std::move` in our object constructors, do you still think it is useful to introduce this change at that time ? 

See https://en.cppreference.com/w/cpp/container/vector/emplace_back

Also after this change, will we have a mix of `push_back` and `emplace_back`, or will `emplace_back` be used everywhere ?

**Comment by hjmjohnson** [comment]
I do think it is useful.  we may not take full advantage of this yet for Slicer objects, it does work for objects that follow the "rule of zero", or other objects.  I see no downsides to using emplace_back in any cases.  Only upsides for object types that support move semantics.

**Pull request merged**